Source/AwsCommonRuntimeKit/io/HostResolver.swift

```diff
@@ -30,26 +28,22 @@ public final class DefaultHostResolver: HostResolver {
         let shutdownCallbackOptionPtr = shutDownOptions?.toShutDownCPointer()
 
         self.shutDownOptions = shutDownOptions
-        let options = aws_host_resolver_default_options(max_entries: maxHosts,
-                                                        el_group: elg.rawValue,
-                                                        shutdown_options: shutdownCallbackOptionPtr,
-                                                        system_clock_override_fn: nil)
-        self.hostResolverOptionsPtr = fromPointer(ptr: options)
-        self.rawValue = aws_host_resolver_new_default(allocator.rawValue, hostResolverOptionsPtr)
-
-        let config = aws_host_resolution_config(
-            impl: aws_default_dns_resolve,
-            max_ttl: maxTTL,
-            impl_data: nil
-        )
-
-        self.config = fromPointer(ptr: config)
 
+        var options = aws_host_resolver_default_options(max_entries: maxHosts,
+                el_group: elg.rawValue,
+                shutdown_options: shutdownCallbackOptionPtr,
+                system_clock_override_fn: nil)
+        rawValue = aws_host_resolver_new_default(allocator.rawValue, &options)
```

Comment:
Swift doesn’t have a pass by reference. &options implicitly converts options to a pointer to pass to the C function. The pointer generated by the ampersand syntax is only valid for the duration of that function call. We can use this when we don’t need long lived pointers. 
Referece: https://developer.apple.com/forums/thread/674633

Reply:
hey look, it's another call to some `_new()` function where we never checked whether it returned NULL